File: 프로그래머스/0/181879. 길이에 따른 연산/길이에 따른 연산.py
def solution(num_list):
    answer = 0
    g =1
    # 리스트의 길이가 11 이상 len(num_list) >= 11
    # 모든 원소 다 더하고
    
    # 리스트의 길이 < 10 
    # 모든 원소의 곱
    
    if  len(num_list) >=11 :
        for num in num_list :
            answer += num
    elif len(num_list) <= 10 :
         for num in num_list :
            g *= num # 곱하기에서 0이 초기값이면 항상 0에 수렴합니다.
            answer = g 
    return answer

File: 프로그래머스/0/181879. 길이에 따른 연산/길이에 따른 연산.js
function solution(num_list) {
    var answer = 0;
    let leng = num_list.length
    let s1 = 0 ; let s2 = 1; 
    
    // 리스트의 길이 11 이상   길이 >= 11 
      // 모든 원소의 합 
      if(leng >= 11){
    
         for(let i = 0 ; i< leng ; i++){
             s1 += num_list[i];
         }
          
          answer = s1;
    // 그게 아니라 길이 10 이하
       // 모든 원소의 곱
      }else if( leng <= 10){
          
          for(let j = 0 ; j < leng ; j++){
              s2 *= num_list[j];
          }
          answer = s2;
      }
    
    
    return answer;
}